quipucords/api/credential/serializer.py

```diff
@@ -42,6 +43,19 @@ class CredentialSerializer(NotEmptySerializer):
     """Base Serializer for the Credential model."""
 
     sources = RelatedSourceSerializer(many=True, read_only=True)
+    auth_type = SerializerMethodField()
+
+    def get_auth_type(self, credential: Credential) -> str:
+        """Determine a credential's authentication type."""
+        if credential.auth_token:
+            return "auth_token"
+        elif credential.password:
+            return "password"
+        elif credential.ssh_key:
+            return "ssh_key"
+        elif credential.ssh_keyfile:
+            return "ssh_keyfile"
+        return "unknown"
```

Comment:
I support these could reference an enum or constants or something, but… effort? 🤷  This may be a short-lived piece of code anyway since we've started working on the v2 credentials api. This just needs to help cover the gap since the new UI code is suffering without it.

Reply:
I am ok with this.